### DISCUSSION TITLE
fix-a-mistake

### DISCUSSION THREAD

**Discussion by sxhthreo** [crytic/slither#2301] [General]
slither/examples/scripts/possible_paths.py#all_function_definitions
There are these codes:
resolve_function in target_function = next( (function for function in contract.functions if function.name == function_name), None )
In the above code, contract.functions needs to be modified to contract.functions_declared. The search scope will be limited to the functions defined by the contract. Otherwise, if there is inheritance, the parent class will be returned.

**Comment by 0xalpharush** [comment]
Do you have an example of an incorrect output that you could share? If the function is virtual, maybe it's not working as expected. Otherwise, if inheritance is used, it is correct to search for the function in parent contracts.

**Comment by sxhthreo** [comment]
Please look this example. Solidity document is as follows:
```
//SPDX-License-Identifier: Unlicense
pragma solidity ^0.8.0;
contract BaseContract{
    
    function f1(uint d) virtual public{
    }
}

contract Contract is BaseContract{
    mapping(address=>uint) balances;


    function f1(uint b) override public{
        address account = address(123);
        balances[account] = 456;
        super.f1(b);
    }

}
```
run the code as  follows:
```
from slither.slither import Slither

def resolve_function(contract_name, function_name):
    """
    Resolves a function instance, given a contract name and function.
    :param contract_name: The name of the contract the function is declared in.
    :param function_name: The name of the function to resolve.
    :return: Returns the resolved function, raises an exception otherwise.
    """
    # Obtain the target contract
    contracts = slither.get_contract_from_name(contract_name)

    # Verify the contract was resolved successfully
    if len(contracts) != 1:
        raise ValueError(f"Could not resolve target contract: {contract_name}")
    contract = contracts[0]
    # Obtain the target function
    target_function = next(
        (function for function in contract.functions if function.name == function_name), None
    )

    # Verify we have resolved the function specified.
    if target_function is None:
        raise ValueError(f"Could not resolve target function: {contract_name}.{function_name}")

    # Add the resolved function to the new list.
    return target_function

source_file = '../example/test_for_test.sol'
destination_file = '../example/process_test1.sol'
slither = Slither(source_file)
parameters_list = resolve_function("Contract", "f1").parameters
parameters_str = ', '.join(str(param) for param in parameters_list)
print('parameters are follows: [' + parameters_str + ']')
```
The command line will output the result as folllows: `parameters are follows: [d]`
We see that the parameters he gets are the parameters of the parent class function, not the subclass. It is because `contract.functions` will include the functions that the parent class have, so we can replace it with `contract.functions_declared`.

**Reply by 0xalpharush** [reply]
Would you like to submit a PR?

**Reply by sxhthreo** [reply]
My pleasure. Here is my PR: https://github.com/crytic/slither/pull/2308
